src/runtime/virtcontainers/clh.go

```diff
@@ -1081,9 +1085,20 @@ func (clh *cloudHypervisor) ResumeVM(ctx context.Context) error {
 
 // StopVM will stop the Sandbox's VM.
 func (clh *cloudHypervisor) StopVM(ctx context.Context, waitOnly bool) (err error) {
+	clh.mu.Lock()
+	defer func() {
+		atomic.StoreInt32(&clh.stopped, 1)
```

Comment:
This means that the VM will always be considered as stopped when `StopVM()` returns, even if something goes wrong in `clh.terminate(ctx, waitOnly)`... this seems incorrect. This state should only be set if we know the VM has actually stopped.

Reply:
You are right, I missed that. The same issue I think applies as well to qemu, stopped is set at the end without checking the error.

Reply:
Fixed in qemu and clh.